crates/iceberg/src/catalog/mod.rs

```diff
@@ -758,23 +760,19 @@ mod tests {
 {
     "action": "add-spec",
     "spec": {
-        "spec-id": 1,
         "fields": [
             {
                 "source-id": 4,
-                "field-id": 1000,
```

Comment:
Why remove these? They are part of the `PartitionField`: https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.py#L131-L135

Reply:
On Java:

https://github.com/apache/iceberg/blob/af9522ac7e8e25dc044622c566b66301b6df9581/core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java#L447-L451

And there it is optional:

https://github.com/apache/iceberg/blob/af9522ac7e8e25dc044622c566b66301b6df9581/core/src/main/java/org/apache/iceberg/PartitionSpecParser.java#L128-L145

Reply:
It looks like it is for historical reasons. For old specs they can be missing, and they'll just auto-increment.

Reply:
Thanks for the reference, I saw the comments now. But I think in the rest api request, it's reasonable to make it optional, and ask the server to assign a meaningful field id to it?